raspibolt/raspibolt
# Monitoring channel.db size in welcome/motd script

#### What

This pull-request adds a few lines to the `20-raspibolt-welcome` script to display the `channel.db` size in the LND section of the 'message of the day' (motd) system overview. (See previous version of the PR with comments [here](https://github.com/raspibolt/raspibolt/pull/776))

### Why

When `channel.db` reaches ~1GB on 32-bit OS, LND experienced issues (e.g. [here](https://github.com/lightningnetwork/lnd/issues/4811){:target="_blank"}). Issues can also be met on 64-bit OSes when reaching larger sizes on low-spec devices (e.g. [here](https://github.com/lightningnetwork/lnd/issues/5705){:target="_blank"}). Therefore, monitoring the growth of the channel.db size is critical.

#### How

The `channel.db` size is obtained in the script using the following command (but see section below for possible improvements):
`du -h /mnt/ext/lnd/data/graph/mainnet/channel.db | awk '{print $1}'`

#### Scope

- [x] minor change to bonus guide

#### Test & maintenance

* How can the changes be tested?
  * The change can be tested by running the modified script on the node. The new line should appear in system overview message in the terminal.
* Is there ongoing maintenance effort? 
  * There is no ongoing maintenance effort, this is a one-off change that should remain valid for the foreseeable future.
* If this is a bonus guide: are you willing to update it from time to time?
  * N/A, see above

#### Ideas for future improvements

For now the variable is a string. Getting an integer instead would allow to create a simple conditional statement to colour-code the line in green, orange and red when below or above a certain size (e.g. <500MB green, 500-900MB orange, >900MB red)

#### Animated GIF (optional)

![db](https://media.giphy.com/media/kPrlykW2TpVU4HWx2O/giphy.gif)
